### PR TITLE
Bump up GCI version.

### LINKED ISSUE #32596: Network throughput issues with latest version of container-vm
The upgraded version of container-vm (gci) has significant performance regressions with localhost networking when compared to its previous versions.

@Amey-D is working on making gci kernel fixes to resolve this issue.

This issue is meant to track the GCI fix from release 1.4 perspective. 

The fix for this issue will result in upgrading the gci version pinned to release 1.4 branch.

ETA: <TBD>
Status: Yellow

### PR DESCRIPTION
**Pull request by Amey-D**
``` release-note
   Upgrading Container-VM base image for k8s on GCE. Brief changelog as follows:
    - Fixed performance regression in veth device driver
    - Docker and related binaries are statically linked
    - Fixed the issue of systemd being oom-killable
```

Fixes #32596

This needs a cherrypick into v1.4 release branch because it is fixing v1.4 release blocking issues. This patch is easy and safe to rollback in case of emergencies.

@vishh can you please review?

Fixes #32596 and many other issues.
cc/ @kubernetes/goog-image  FYI

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32738)

### CODE REVIEW

**Comment by vishh** [comment]
Yayy!! Thanks for this much awaited PR @Amey-D

**Comment by vishh** [comment]
I wish the per-pr tests were up and running for GCI, but @mtaufen is working on them. For now, I would rely on the node e2e to pass to validate this PR.

**Comment by vishh** [comment]
@Amey-D Can you add a release note section to the PR summarizing the changes to GCI that is included in this PR?

**Comment by dchen1107** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[cluster/gce/config-default.sh, line 41 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32738#-KRewtVIJOYPlwsY9464:-KRewtVIJOYPlwsY9465:bf6896q) ([raw file](https://github.com/kubernetes/kubernetes/blob/83123c408b53c4eb5b875cf7486cf8519132ed5d/cluster/gce/config-default.sh#L41)):*
> ```Shell
> NODE_OS_DISTRIBUTION=${KUBE_NODE_OS_DISTRIBUTION:-${KUBE_OS_DISTRIBUTION:-gci}}
> CVM_VERSION=container-v1-3-v20160604
> GCI_VERSION="gci-dev-55-8801-0-0"
> ```

Any reason we change milestone from 54 to 55 here? My concern is that milestone 55 includes more changes than what I expected here. 

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32738)*

**Comment by Amey-D** [comment]
Updated commit message with a brief changelog.

**Comment by Amey-D** [comment]
*[cluster/gce/config-default.sh, line 41 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32738#-KRewtVIJOYPlwsY9464:-KRf-JkWa6Toxni_u95Q:bgd3dm2) ([raw file](https://github.com/kubernetes/kubernetes/blob/de891b65e8fe1587597bc622248f8ecb12132ba4/cluster/gce/config-default.sh#L41)):*
<details><summary><i>Previously, dchen1107 (Dawn Chen) wrote…</i></summary>
> Any reason we change milestone from 54 to 55 here? My concern is that milestone 55 includes more changes than what I expected here. 
</details>
Several fixes relevant to k8s were added to 55.  We could backport to 54 but I don't see the real difference because there are no major changes between 54 and 55 (happy to show you the detailed changelog offline :).  Besides it saves us the time to backport in 55 and wait for the build / release process.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/32738)*

**Comment by vishh** [comment]
@dchen1107 If you have any issues with this PR, feel free to remove the lgtm label and continue the conversation.

**Comment by dchen1107** [comment]
Removed lgtm from pr to resolve milestone issue here.

**Comment by dchen1107** [comment]
Here is my concern and why removing lgtm: We have tested everything on GCI with milestone 54. Now at the last minute of cutting release, we switch to milestone 55 which includes tons of other unknown patches (to me). Can we cherrypick those patches affecting Kubernetes to milestone 54?

**Comment by eparis** [comment]
Without agreement and rapid movement from @vishh and @dchen1107 this will not make 1.4.0.

I see the initial comment says this 'fixes release blocker issues'. That is inadequate. Please enumerate exactly what issue(s) you believe is a release blocker and why you believe it should be a release blocker. I apologize but if I don't understand all the issues, it isn't going into the release.

**Comment by Amey-D** [comment]
@k8s-bot gci test this

**Comment by dchen1107** [comment]
@eparis Switching to GCI by default on GCE shouldn't be the blocker for 1.4 release. But if we decided to switch to GCI image on GCE by default (pr was merged before the feature freeze date), this pr is the blocker since the new image with milestone 55 contains a lot of fixes we found on the old image (54). The reason I removed lgtm is that 

* New image has a lot of other changes besides necessary patches.
* There is no enough test results and soak time to convince me it pass our qualification standard. @vishh and gci-image team are working on this.
* With some limited tests, there are some failures requiring the triage, understanding and fix. 

In summary, we are going to decide if switching to GCI image on GCE by default shortly.

**Comment by dchen1107** [comment]
Here is our decision: 

* We are going not to switch GCI by default on GCE for 1.4.0 release. https://github.com/kubernetes/kubernetes/pull/32913 to fix this. 
* Based on above decision, I am going to unblock this pr because it contains the right fix for GCI. 
* Based on above decision, I am ok to cherrypick this release to 1.4 release branch so that the users can try with new ContainerVM image (GCI). @pwittrock, 1.4 release czar!

cc/ @kubernetes/goog-image @kubernetes/goog-image @fabioy @maisem

**Comment by dchen1107** [comment]
LGTM

**Comment by pwittrock** [comment]
@eparis I talked with @dchen1107 and @vishh about this.  GCI will not be enabled by default, and this is something users have to opt-in to.  Since using GCI is opt-in, and not something I am validating specifically as part of the OSS release, IMHO it is fine to let the GCI team set whatever image they want to use.  A better solution in the future would be do decouple the set of base machine images that can be used from the release altogether so that they can be updated independently from one another.

**Comment by dchen1107** [comment]
We are working on those known issues of GCI GCE e2e.

**Comment by pwittrock** [comment]
@k8s-bot test this issue: #IGNORE

**Comment by vishh** [comment]
@Amey-D can you update the GCI image to include the recent fixes?

**Comment by Amey-D** [comment]
@k8s-bot gci test this

**Comment by Amey-D** [comment]
@k8s-bot gci test this

**Comment by Amey-D** [comment]
@k8s-bot gce e2e test this

**Comment by vishh** [comment]
@k8s-bot kubemark test this

**Comment by vishh** [comment]
@pwittrock this is a candidate for manual merging since submit queue is not yet setup to catch GCI issues

**Comment by vishh** [comment]
Yayy. Thanks Amey!

Can you also post a cherry-pick to v1.4? Run `hack/cherry_pick_pull.sh upstream/release-1.4 32738`
